apps/SageAccountWeb/src/hooks/useSkipMfaPrompt.test.ts

```diff
@@ -0,0 +1,251 @@
+import { renderHook, act } from '@testing-library/react'
+import { vi, describe, it, expect, beforeEach, afterEach } from 'vitest'
+import dayjs, { Dayjs } from 'dayjs'
+import { when } from 'vitest-when'
+import { useSkipMfaPrompt } from './useSkipMfaPrompt'
+import { useLocalStorageValue } from '@react-hookz/web'
+
+vi.mock('@react-hookz/web', () => ({
+  useLocalStorageValue: vi.fn(),
+}))
+
+vi.mock('dayjs', async importOriginal => {
+  return {
+    ...(await importOriginal()),
+    default: vi.fn(() => ({
+      subtract: vi.fn().mockReturnThis(),
+      isAfter: vi.fn(),
+    })),
+  }
+})
+
+const mockUseLocalStorageValue = vi.mocked(useLocalStorageValue)
+const mockDayjs = vi.mocked(dayjs)
+
+describe('useSkipMfaPrompt', () => {
+  const mockSetTimestamp = vi.fn()
+  const mockUseLocalStorageValueReturn = {
+    value: null,
+    set: mockSetTimestamp,
+    remove: vi.fn(),
+    fetch: vi.fn(),
+  }
+  // The dayjs mock for the current time should always return a consistent instance
+  const mockDayjsInstanceForCurrentTime = {
+    subtract: vi.fn().mockReturnThis(),
+  }
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+    // Mock console.error to suppress error logs in tests
+    vi.spyOn(console, 'error').mockImplementation(() => {})
+
+    when(mockDayjs)
+      .calledWith() // No arguments for the current time
+      .thenReturn(mockDayjsInstanceForCurrentTime as unknown as Dayjs)
+  })
+
+  afterEach(() => {
+    vi.restoreAllMocks()
+  })
+
+  describe('hasSkippedRecently', () => {
+    it('should return false when no timestamp is stored', () => {
+      mockUseLocalStorageValue.mockReturnValue({
+        ...mockUseLocalStorageValueReturn,
+        value: null,
+      })
+
+      const { result } = renderHook(() => useSkipMfaPrompt())
+
+      expect(result.current.hasSkippedRecently).toBe(false)
+    })
+
+    it('should return false when timestamp is empty string', () => {
+      mockUseLocalStorageValue.mockReturnValue({
+        ...mockUseLocalStorageValueReturn,
+        value: '',
+      })
+
+      const { result } = renderHook(() => useSkipMfaPrompt())
+
+      expect(result.current.hasSkippedRecently).toBe(false)
+    })
+
+    it('should return true when timestamp is less than 24 hours old', () => {
+      const mockTimestamp = '2023-12-01T10:00:00.000Z'
+      mockUseLocalStorageValue.mockReturnValue({
+        ...mockUseLocalStorageValueReturn,
+        value: mockTimestamp,
+      })
+
+      const mockDayjsInstanceForStoredTimestamp = {
+        isAfter: vi.fn().mockReturnValue(true),
+      }
+
+      when(mockDayjs)
+        .calledWith(mockTimestamp)
+        .thenReturn(mockDayjsInstanceForStoredTimestamp as unknown as Dayjs)
```

Comment:
I thought using `vitest-when` to write a declarative/mockito-style mock was easier to write/read here. Just seems so much less error-prone & forces you to think about how the mock is being used compared to writing imperative conditional logic in a mock.